### PR TITLE
coost: 3.0.0 -> 3.0.2, migrate to by-name

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/co/coost/package.nix [hunk 1]
```diff
@@ -34,6 +34,7 @@ stdenv.mkDerivation rec {
     "-DBUILD_SHARED_LIBS=ON"
   ] ++ lib.optional withCurl "-DWITH_LIBCURL=ON" ++ lib.optional withOpenSSL "-DWITH_OPENSSL=ON";
 
+  output = [ "out" "dev" ];
```
Comment (NickCao) [inline pkgs/by-name/co/coost/package.nix:37]: outputs?

Comment (Sigmanificient) [reply pkgs/by-name/co/coost/package.nix]: Oh yeah, you are right

File: pkgs/by-name/co/coost/package.nix [hunk 2]
```diff
@@ -10,15 +10,15 @@
...
   pname = "coost";
-  version = "3.0.0";
+  version = "3.0.2";
 
   src = fetchFromGitHub {
     owner = "idealvin";
-    repo = pname;
-    rev = "v${version}";
-    sha256 = "sha256-qpJh1yl0lYYszNHGo5Jkbzal2hnVzg7UUxiyg/Grva8=";
+    repo = finalAttrs.pname;
```
Comment (kirillrdy) [inline pkgs/by-name/co/coost/package.nix:19]: https://github.com/NixOS/nixpkgs/issues/310373

File: pkgs/by-name/co/coost/package.nix [hunk 3]
```diff
@@ -10,15 +10,15 @@
...
+    rev = "v${finalAttrs.version}";
+    sha256 = "sha256-HbMenAL/UWsqQ1o7cMeWfwXkLh4GxIKV7iuZQD3hDA8=";
```
Comment (kirillrdy) [inline pkgs/by-name/co/coost/package.nix:21]:
```suggestion
    hash = "sha256-HbMenAL/UWsqQ1o7cMeWfwXkLh4GxIKV7iuZQD3hDA8=";
```

File: pkgs/by-name/co/coost/package.nix (hunk 2 above)
Comment (Sigmanificient) [reply pkgs/by-name/co/coost/package.nix:19]: not sure what you want me to do

Comment (eclairevoyant) [reply pkgs/by-name/co/coost/package.nix:19]:
```suggestion
    repo = "coost";
```

`repo = finalAttrs.pname;` is an antipattern

**Review by wineee — APPROVED** [review]

**Pull request merged**